src/commands/dsyms/upload.ts

```diff
@@ -67,11 +67,12 @@ export class UploadCommand extends Command {
       datadogSite: this.config.datadogSite,
       metricsLogger: metricsLogger.logger,
     })
-    const payloads = await getMatchingDSYMFiles(searchPath)
+    const payloads = await getMatchingDSYMFiles(searchPath, this.context)
+    const validPayloads = payloads.filter((payload) => payload !== undefined) as Dsym[]
```

Comment:
It is not necessary to cast as `Dsym[]` as typescript infers it correctly
```suggestion
    const validPayloads = payloads.filter((payload) => payload !== undefined)
```

Reply:
maybe this is a problem with my local but it can't infer properly and gives me an error without casting.
is this a blocker issue? if so, i can try to investigate further

Reply:
This not a blocking issue, feel free to discard!